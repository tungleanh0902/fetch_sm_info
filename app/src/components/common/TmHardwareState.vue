<template>
  <div class="tm-hardware-state">
    <div class="tm-hardware-state-content">
      <img
        v-if="loading"
        src="~assets/images/loader.svg"
        alt="a small spinning circle to display loading"
        class="tm-hardware-state__icon"
      />
      <i v-else-if="icon" class="tm-hardware-state__icon material-icons">
        {{ icon }}
      </i>
      <div class="tm-hardware-state__label">
        <slot />
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    icon: {
      type: String,
      default: ``,
    },
    loading: {
      type: Boolean,
      default: false,
    },
  },
}
</script>

<style scoped>
.tm-hardware-state {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: var(--app-fg);
  margin: 1.5rem auto 0;
  padding: 1.5rem;
  user-select: none;
  border-radius: 0.25rem;
}

.tm-hardware-state-content {
  display: flex;
  align-items: center;
}

.tm-hardware-state__icon {
  font-size: 1.5rem;
  padding-right: 1rem;
  color: var(--dim);
}

.tm-hardware-state__label {
  color: var(--bright);
  margin-right: 1rem;
}
</style>
