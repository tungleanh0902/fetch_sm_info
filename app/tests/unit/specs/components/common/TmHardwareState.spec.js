import { shallowMount } from "@vue/test-utils"
import TmHardwareState from "common/TmHardwareState"

describe(`TmHardwareState`, () => {
  let wrapper

  const propsData = {
    icon: `rotate_right`,
    loading: true,
  }

  beforeEach(() => {
    wrapper = shallowMount(TmHardwareState, {
      propsData,
      slots: {
        default: `Detecting your Ledger Wallet`,
      },
    })
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`has a label`, () => {
    expect(wrapper.find(`.tm-hardware-state__label`).text().trim()).toContain(
      `Detecting your Ledger Wallet`
    )
  })
})
