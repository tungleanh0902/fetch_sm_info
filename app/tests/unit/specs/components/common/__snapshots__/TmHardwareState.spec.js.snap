// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmHardwareState has the expected html structure 1`] = `
<div
  class="tm-hardware-state"
>
  <div
    class="tm-hardware-state-content"
  >
    <img
      alt="a small spinning circle to display loading"
      class="tm-hardware-state__icon"
      src="~assets/images/loader.svg"
    />
     
    <div
      class="tm-hardware-state__label"
    >
      Detecting your Ledger Wallet
    </div>
  </div>
</div>
`;
